httpie/input.py

```
@@ -0,0 +1,693 @@
+            if item.sep in SEP_GROUP_DATA_EMBED_ITEMS:
+                try:
+                    with open(os.path.expanduser(value), 'rb') as f:
+                        value = f.read().decode('utf8')
+                except IOError as e:
+                    raise ParseError('"%s": %s' % (item.orig, e))
+                except UnicodeDecodeError:
+                    raise ParseError(
+                        '"%s": cannot embed the content of "%s",'
+                        ' not a UTF8 or ASCII-encoded text file'
+                        % (item.orig, item.value)
+                    )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)